pkg/graphql/request.go

```diff
@@ -2,17 +2,40 @@ package graphql
 
 import (
 	"encoding/json"
+	"errors"
 	"io"
+	"io/ioutil"
 )
 
+var ErrEmptyRequest = errors.New("the provided request is empty")
+
 type Request struct {
 	OperationName string          `json:"operation_name"`
 	Variables     json.RawMessage `json:"variables"`
 	Query         string          `json:"query"`
 }
 
 func UnmarshalRequest(reader io.Reader) (*Request, error) {
```

Comment:
why not "request *Request" as second argument? Gives the user control over creating objects.

Reply:
sounds good - this is would feel like using json.Unmarshal